torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -296,6 +301,10 @@ def _affine_bounding_box_xyxy(
 
     if shear is None:
         shear = [0.0, 0.0]
+    elif isinstance(shear, (int, float)):
+        shear = [float(shear), 0.0]
+    elif len(shear) == 1:
+        shear = [shear[0], shear[0]]
```

Comment:
For the changes on the body of this method, I was tempted to reuse `_affine_parse_args()` but that would probably slow things down, so I opted for porting only the absolutely necessary changes.

Reply:
All these can be avoided if we decide to return meta-data from the kernels. But I would take this as a separate discussion and follow up PRs.

Reply:
I used it to align the parsing with all other `affine_*` kernels. By using this function we only have a single point for the parsing for all kernels. Unless we detect a bottleneck there, I think that outweighs minimal speed ups.

I've reverted some changes on my PR in 6dfc9657ce89fe9e018a11ee25a8e26c7d3d43c6. Be aware that this is the revert commit, i.e. the diff is inverted.

Apart from using the function, I also refactored it a little to avoid the `None` defaults for most arguments. They are only there since we are also calling this function from `rotate_bounding_box` and there we only have the angle. I opted to simply pass the the current default values explicitly from `rotate_bounding_box` so that `_affine_parse_args()` does not have default values.

Reply:
As discussed offline, I agree with your changes and I should probably examine bringing them here once you merge the test changes. :)

Reply:
This is still a TODO, right?

Reply:
done